src/main/java/softdesign/Environment.java

```diff
@@ -1,44 +1,65 @@
 package main.java.softdesign;
 
+import main.java.softdesign.map.Map;
+import main.java.softdesign.map.CartesianGridMap;
+import main.java.softdesign.map.CartesianCoordinate;
+
 import simbad.sim.Box;
 import simbad.sim.EnvironmentDescription;
 import simbad.sim.Wall;
 
 import javax.vecmath.Color3f;
 import javax.vecmath.Vector3d;
 import javax.vecmath.Vector3f;
+import javax.vecmath.Point3d;
 import java.awt.*;
 import java.util.concurrent.ThreadLocalRandom;
 
 public class Environment extends EnvironmentDescription {
 
-	private static final Environment INSTANCE = new Environment();
+	private static final int SMALL = 5;
+	private static final int MEDIUM = 15;
+	private static final int LARGE = 25;
+
+	private static final float BOX_SIZE = 1f;
+
+	private static final Environment INSTANCE = new Environment(MEDIUM);
 
-	public static final int SIZE = 21;
-	public static final int TOTAL_NUMBER_OF_POINTS = SIZE * SIZE;
+	private int size;
+	private Map map;
```

Comment:
`Environment` should have no knowledge of `Map`. It should not even create a `Map`. Why is an environment mapping itself? It makes absolutely no sense. I get that it is convenient but that doesn't mean you can throw all logic out of the window.

I once created a method `CentralStation#drawMap` (which you deleted). Please move the logic for drawing the map there.

Reply:
So this is because the environment creates obstacles, which need to be placed on the map to. I thought about this a bot, and this is the best I came up with

Reply:
Else we have to move all the environment logic (like obstaces) into central station. Ugh.

Reply:
This point has been discussed on Slack.